client/service/src/client/call_executor.rs

```diff
@@ -81,15 +83,23 @@ where
 		Block: BlockT,
 		B: backend::Backend<Block>,
 	{
-		let code = self.wasm_override
+		let code = if let Some(d) = self.wasm_override
 			.as_ref()
 			.map::<sp_blockchain::Result<Option<RuntimeCode>>, _>(|o| {
 				let spec = self.runtime_version(id)?.spec_version;
 				Ok(o.get(&spec, onchain_code.heap_pages))
 			})
 			.transpose()?
-			.flatten()
-			.unwrap_or(onchain_code);
+			.flatten() {
+			debug!(target: "wasm_overrides", "using WASM override for block {}", id);
+			d
+		} else {
+			debug!(target: "wasm_overrides",
```

Comment:
I'm not sure how useful this debug! line is, most of the time overrides won't be used, so this  will just be taking up space. 

IMO the line that tells us we are using an override is enough, if that line isn't printed we can assume that the override was not used